src/sets.jl

```diff
@@ -438,6 +438,18 @@ Base.isapprox(a::SOS2{T}, b::SOS2{T}; kwargs...) where T = isapprox(a.weights, b
 
 dimension(s::Union{SOS1, SOS2}) = length(s.weights)
 
+"""
+Set of (x,y) that satisfy the indicator constraint:
+    y ∈ 𝔹, y = 1 ==> a^T x ∈ S
+
+S can be one of `SmallerThan, GreaterThan, EqualTo`
+"""
+struct IndicatorSet{T <: Real, S <:Union{SmallerThan,GreaterThan,EqualTo}} <: MOI.AbstractVectorSet
+    a::Vector{T}
+    s::S
+end
```

Comment:
Another approach that might work is
```julia
"""
    IndicatorSet(y::VariableIndex, set::AbstractSet)

The set `{y \in {0, 1}, y=1 ==> S, y=0 ==> R^n}`.
"""
struct IndicatorSet{S}
    y::MOI.VariableIndex
    set::S
end
```
Then solvers could implement
```julia
add_constraint(::ModelLike, ::ScalarAffineFunction, ::IndicatorSet)
```

Reply:
Ok I see, I had a mental model of the set being independent from the variables.
Another one I was thinking of was using an `IndicatorFunction` with one index for y and the `ScalarAffineTerms` forx

Reply:
Although `VectorOfVariables-in-IndicatorSet` is also good. Should `y` already be binary? Or is it the solvers job to add this constraint when an indicator is applied.

Reply:
> I had a mental model of the set being independent from the variables

I'm not sure if we ever settled on this.

Reply:
the pain with the set-based design and a vector function is that the user must remember to put the binary first, far from ideal

Reply:
It is somewhat consistent with the cones requiring the user to put the epigraph variable first: https://github.com/JuliaOpt/MathOptInterface.jl/issues/547

Reply:
I think the current proposal is more easily generalized to a disjunction of linear inequalities, that is `ax ≤ b v cx ≤ d`, where `a` and `c`  with `b` and `d` could be stored in the `Set` while using a `Vector` function.